### PR TITLE
melodeon: init at 0.4.1

### PR DESCRIPTION
**Pull request by edgar-vincent**
## Description of changes

This adds melodeon (https://github.com/CDrummond/melodeon), a web app wrapper around MaterialSkin, a theme for Lyrion (https://lyrion.org, formerly known as Logitech Media Server and packaged as `slimserver` in nixpkgs).
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by edgar-vincent** [comment]
Thanks for your patience and sorry for all the noise, I'm new at this.

File: pkgs/by-name/me/melodeon/package.nix [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "melodeon";
+  version = "0.4.1";
+
+  src = fetchFromGitHub {
+    owner = "CDrummond";
+    repo = "melodeon";
+    rev = "refs/tags/${finalAttrs.version}";
+    sha256 = "sha256-k74iJ3tmkMGUh4f538tjD7cv4ZXxm3x6M/d5HU25yPQ=";
```
Comment (Aleksanaa) [inline pkgs/by-name/me/melodeon/package.nix:18]:
```suggestion
    hash = "sha256-k74iJ3tmkMGUh4f538tjD7cv4ZXxm3x6M/d5HU25yPQ=";
```

File: pkgs/by-name/me/melodeon/package.nix [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  };
+
+  buildInputs = [
+    qt6.qtbase
+    qt6.qtwebengine
+  ];
+
+  nativeBuildInputs = [
+    cmake
+    gcc
```
Comment (Aleksanaa) [inline pkgs/by-name/me/melodeon/package.nix:29]:
```suggestion
```

File: pkgs/by-name/me/melodeon/package.nix [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    qt6.wrapQtAppsHook
+  ];
+
+  cmakeFlags = [ "-DCMAKE_BUILD_TYPE=Release" ];
+
+  meta = with lib; {
```
Comment (Aleksanaa) [inline pkgs/by-name/me/melodeon/package.nix:35]:
```suggestion
  meta = {
```

File: pkgs/by-name/me/melodeon/package.nix [hunk 4]
```diff
@@ -0,0 +1,44 @@
+{
+  stdenv,
+  lib,
+  fetchFromGitHub,
+  cmake,
+  gcc,
```
Comment (Aleksanaa) [inline pkgs/by-name/me/melodeon/package.nix:6]:
```suggestion
```

**Comment by edgar-vincent** [comment]
Thanks a lot to both of you.

**Pull request merged**